lib/common/containers/WatchButton.js

```diff
@@ -18,16 +22,58 @@ type ContainerProps = {
 }
 
 type Props = ContainerProps & {
+  resendEmailConfirmation: typeof resendEmailConfirmation,
+  setErrorMessage: typeof setErrorMessage,
   updateTarget: typeof updateTargetForSubscription
 }
 
 class WatchButton extends Component<Props> {
   messages = getComponentMessages('WatchButton')
 
   _onToggleWatch = () => {
-    const {updateTarget, user, target, subscriptionType} = this.props
-    if (user.profile) updateTarget(user.profile, target, subscriptionType)
-    else console.warn('User profile not found. Cannot update subscription', user)
+    const {
+      isWatching,
+      resendEmailConfirmation,
+      setErrorMessage,
+      subscriptionType,
+      target,
+      updateTarget,
+      user
+    } = this.props
+    if (user.profile) {
+      if (!isWatching && !user.profile.email_verified) {
+        if (confirm(this.messages('emailVerificationConfirm'))) {
+          resendEmailConfirmation(user)
+            .then((result: boolean) => {
```

Comment:
I was under the impression that you were somewhat opposed to using thunk in this way (chaining actions like this). The other way to handle this would be to just interact with the redux store. I don't have hugely strong feelings about this, but was under the impression that this was the direction we were heading.

EDIT: On second thought, this approach is growing on me -- it's so ephemeral that putting the result in the store seems silly and I like that you're using `setErrorMessage` in the component so that we can set the message based on `WatchButton` messages.

Reply:
Yeah, it sorta seemed easier to do it this way since messages apply only to components, otherwise this logic would all be in an action.